src/lib/Libutil/pbs_aes_encrypt.c

```diff
@@ -150,3 +152,101 @@ pbs_decrypt_pwd(char *crypted, int credtype, size_t len, char **uncrypted)
 
         return 0;
 }
+
+/**
+ * @brief
+ *	encode_to_base64 - Encode data into bas64 format
+ *
+ * @param[in]		buffer			-	Data buffer for encoding
+ * @param[in]		buffer_len		-	Length of the data buffer
+ * @param[in/out]	ret_encoded_data	-	Return the encoded data
+ *
+ * @return int
+ * @retval 0 - for success
+ * @retval 1 - for failure
+ */
+int
+encode_to_base64(const unsigned char* buffer, size_t buffer_len, char** ret_encoded_data)
+{
+	BUF_MEM *b_mem;
+	BIO *mem_obj1, *mem_obj2;
+	int buf_len = 0;
+
+	mem_obj1 = BIO_new(BIO_s_mem());
+	mem_obj2 = BIO_new(BIO_f_base64());
+
+	mem_obj1 = BIO_push(mem_obj2, mem_obj1);
+	BIO_set_flags(mem_obj1, BIO_FLAGS_BASE64_NO_NL);
+	BIO_write(mem_obj1, buffer, buffer_len);
+	(void)BIO_flush(mem_obj1);
+	BIO_get_mem_ptr(mem_obj1, &b_mem);
+
+	*ret_encoded_data = (*b_mem).data;
+	buf_len = (*b_mem).length;
+
+	*ret_encoded_data = malloc(buf_len + 1);
+	if (*ret_encoded_data == NULL) {
+		BIO_free_all(mem_obj1);
+		return (1);
+	}
+	memcpy(*ret_encoded_data, (*b_mem).data, buf_len);
+	(*ret_encoded_data)[buf_len] = '\0';
+
+	BIO_free_all(mem_obj1);
```

Comment:
Why is mem_obj2 not freed?

Reply:
BIO_free_all() call frees up an entire BIO chain